api/types/provisioning.go

```diff
@@ -704,3 +720,28 @@ func (a *ProvisionTokenSpecV2GCP) checkAndSetDefaults() error {
 	}
 	return nil
 }
+
+func (a *ProvisionTokenSpecV2Spacelift) checkAndSetDefaults() error {
+	if a.Hostname == "" {
+		return trace.BadParameter(
+			"hostname: should be set to the hostname of the spacelift tenant",
+		)
+	}
+	if strings.Contains(a.Hostname, "/") {
+		return trace.BadParameter(
+			"hostname: should not contain the scheme or path",
+		)
+	}
```

Comment:
The error message indicates that the scheme (protocol) should not be included. The documentation should clear up what folks are expected to place here. I'm generally a little cautious of us modifying a value specified by the user in case we accidentally do the wrong thing and create an even more opaque error.